### ISSUE TITLE
Reorganize the advanced section

### ISSUE THREAD

**Issue by applemethod** [cpinitiative/usaco-guide#898]
I found https://usaco.guide/adv/dp-more?lang=cpp and https://usaco.guide/adv/persistent?lang=cpp extremely useful (especially with the multiple parts) so I was thinking maybe you could split it into multiple modules?

Especially with the advanced dp optimizations, I thought dp on broken profile, connected component dp, and knuth's optimization were all useful in and of themselves, probably worthy of a new module. Especially some of the other modules that I looked at, some of them were extremely contrived, so I thought we could have more dp modules and less data structure modules (after all I also find that dp is extremely important and the other data structures weren't used as much)

Nevertheless, I feel like these dp optimizations are extremely important and deserve more attention.

**Comment by huynhtrankhanh** [comment]
The "Max Suffix Query with Insertions Only" technique for example is unlikely to be useful outside of the Springboards problem. Also if we don't have an explanation available for a certain topic, we should maybe provide more links to helpful resources so that future contributors can help expand the articles.

**Comment by bqi343** [comment]
> Especially some of the other modules that I looked at, some of them were extremely contrived
???

**Comment by applemethod** [comment]
> are you suggesting that some modules be removed
maybe not remove them, but maybe merging similar ones (like maybe merging "counting minimums with a segtree" into the plat section with "more applications of segtree")

**Comment by bqi343** [comment]
Haven't seen counting minimums in platinum though, so a bit hesitant to put them together

**Comment by thecodingwizard** [comment]
is this a stale issue? (should I close it?)

**Comment by bqi343** [comment]
splitting DP opts is probably a good idea (though I'm too lazy to split them / add content myself :/)

**Comment by SansPapyrus683** [comment]
so the verdict is just "hey it'd be nice to split DP opts"?

**Comment by SansPapyrus683** [comment]
bump?

**Comment by bqi343** [comment]
Yes, "DP on Broken Profile" should probably go in its own module.
